pkg/syncer/server.go

```diff
@@ -129,38 +143,25 @@ func (s *RegionSyncer) RunServer(ctx context.Context, regionNotifier <-chan *cor
 			return
 		case first := <-regionNotifier:
 			failpoint.InjectCall("syncRegionChannelFull")
-			requests = append(requests, first.GetMeta())
-			stats = append(stats, first.GetStat())
-			// bucket should not be nil to avoid grpc marshal panic.
-			bucket := &metapb.Buckets{}
-			if b := first.GetBuckets(); b != nil {
-				bucket = b
-			}
-			buckets = append(buckets, bucket)
-			leaders = append(leaders, first.GetLeader())
+
+			processRegion(first)
 			startIndex := s.history.getNextIndex()
-			s.history.record(first)
-			pending := len(regionNotifier)
-			for i := 0; i < pending && i < maxSyncRegionBatchSize; i++ {
-				region := <-regionNotifier
-				requests = append(requests, region.GetMeta())
-				stats = append(stats, region.GetStat())
-				// bucket should not be nil to avoid grpc marshal panic.
-				bucket := &metapb.Buckets{}
-				if b := region.GetBuckets(); b != nil {
-					bucket = b
+		loop:
+			for range maxSyncRegionBatchSize {
+				select {
+				case region := <-regionNotifier:
+					processRegion(region)
+				default:
+					break loop
 				}
-				buckets = append(buckets, bucket)
-				leaders = append(leaders, region.GetLeader())
-				s.history.record(region)
 			}
 			regions := &pdpb.SyncRegionResponse{
 				Header:        &pdpb.ResponseHeader{ClusterId: keypath.ClusterID()},
-				Regions:       requests,
+				Regions:       slices.Clone(requests),
```

Comment:
why add clone

Reply:
This is the code I added when I tried to call broadcast asynchronously, but I ultimately did not use the asynchronous broadcast. I will revert them.